cluster.go

```diff
@@ -358,6 +380,94 @@ func parseShards(shards RedisMessage, defaultAddr string, tls bool) map[string]g
 	return groups
 }
 
+func (c *clusterClient) runClusterTopologyRefreshment() {
+	ticker := time.NewTicker(c.opt.ClusterTopologyRefreshmentOption.ScanInterval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-c.stopCh:
+			return
+		case <-ticker.C:
+			c.conditionalRefresh()
```

Comment:
Yes, now there is no duplicated `getClusterTopology` at the end. However, I still want all the refresh processes, including the new conditional one, to be gated by the `lazyRefresh` function to ensure at most one `getClusterTopology` per second and not to flood the servers. `lazyRefresh` serves the same purpose as your initial cool-down period proposal. Can we merge the `_refresh` and `conditionalRefresh` and then just use `lazyRefresh` here?
```suggestion
			c.lazyRefresh()
```

Reply:
[826de2c](https://github.com/redis/rueidis/pull/638/commits/826de2c76a7906b25727ee529ba91d28622d6457)

just call lazyRefresh

Reply:
Yes, I am ok with this one, but why do you delete your detection logic?

Reply:
Oh, i thought you want to use original one. i changed [9fd1da9](https://github.com/redis/rueidis/pull/638/commits/9fd1da9a6bff4f2cf0e05e77ca66135c9aa50250)

Reply:
[fa0b58a](https://github.com/redis/rueidis/pull/638/commits/fa0b58a22e8af7dc63d426d4bd0423c4c8182e25)

We must not merge `_refresh` and `conditionalRefresh`.

`_refresh` is called,
1. In a constructor. This case must init all conns, pslots, rslots. so should be all logic must be called.
2. When got error from redis cluster. This case must be refresh even if cluster is not modified. because a connection is band.

Reply:
Would 1 be an issue if we made `_refresh` conditionally? I thought in that case the checking logic would be bypassed.

For 2 and 3, I don't get your points of not making `_refresh` conditionally. They are not related to the refresh process but are more related to how we act on the result of it.

Reply:
[c02c256](https://github.com/redis/rueidis/pull/638/commits/c02c2562a7276d3bf265651534aab1e7403bf0cf)